### PR TITLE
Improve markdown formatting and fix pipe in table

### PR DESCRIPTION
**Pull request by mavimo**
## Changes

 - Improve code formatting in template by creating `codeSnippet`, `codeBlock` and `codeSnippetInTable` filters
 - Added tests for `codeSnippet`, `codeBlock` and `codeSnippetInTable` filters
 - Fixed issue for pipe in code snippet embedded in table (see https://github.com/github/markup/issues/1078 )

Replace https://github.com/stackrox/kube-linter/pull/106

### CODE REVIEW

File: internal/command/common/markdown.go [hunk 1]
```diff
@@ -12,8 +13,14 @@ import (
 func MustInstantiateTemplate(templateStr string, customFuncMap template.FuncMap) *template.Template {
 	tpl, err := template.New("").Funcs(sprig.TxtFuncMap()).Funcs(
 		template.FuncMap{
-			"backtick": func() string {
-				return "`"
+			"codeSnippet": func(code string) string {
+				return "`" + code + "`"
+			},
+			"codeSnippetInTable": func(code string) string {
+				return "`" + strings.Replace(code, "|", "\\|", -1) + "`"
```
Comment (kreamkorokke) [inline internal/command/common/markdown.go:20]: Please consider using \`\\|\` instead of "\\|" to avoid double escaping

**Review by kreamkorokke — COMMENTED** [review]
Awesome work! Please address the lint failure and also checkin the re-generated srcs to this commit as well. Thank you for you contribution.

**Comment by mavimo** [comment]
@kreamkorokke done :)

File: internal/command/common/markdown_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	}
+
+	for _, tt := range templateTests {
+		tpl := common.MustInstantiateTemplate(tt.in, nil)
+
+		var b bytes.Buffer
+
+		tpl.Execute(&b, tt.data)
+
+		if b.String() != tt.out {
```
Comment (kreamkorokke) [inline internal/command/common/markdown_test.go:32]: Sorry that I missed this before: please consider using `assert.Equal` in tests to check for equality.

**Comment by kreamkorokke** [comment]
Sorry that I missed something before. One more minor comment

File: internal/command/common/markdown_test.go (hunk 2 above)
Comment (mavimo) [reply internal/command/common/markdown_test.go]: Np, done 👍

**Pull request merged**